File: _legacy_archive/legacy_ref/atlas/native/ios/SmartSpectraModule.swift
/**
 * Presage SmartSpectra SDK - Native iOS Module
 * 
 * This Swift file needs to be added to your iOS project when you eject from Expo.
 * It bridges the SmartSpectraSwiftSDK to React Native.
 * 
 * SETUP INSTRUCTIONS:
 * 1. Eject from Expo: npx expo prebuild
 * 2. Add SmartSpectra Swift Package: 
 *    - In Xcode: File → Add Package Dependencies
 *    - URL: https://github.com/Presage-Security/SmartSpectra
 *    - Branch: main
 * 3. Copy this file to ios/aac/SmartSpectraModule.swift
 * 4. Add camera permission to Info.plist if not already present
 * 5. Build and run
 */

/*
import Foundation
import React
import SmartSpectraSwiftSDK
import Combine

@objc(SmartSpectraModule)
class SmartSpectraModule: RCTEventEmitter {
    
    private var sdk = SmartSpectraSwiftSDK.shared
    private var vitalsProcessor = SmartSpectraVitalsProcessor.shared
    private var cancellables = Set<AnyCancellable>()
    private var isInitialized = false
    
    override static func moduleName() -> String! {
        return "SmartSpectraModule"
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    override func supportedEvents() -> [String]! {
        return ["onMetricsUpdate", "onStatusChange", "onEdgeMetricsUpdate"]
    }
    
    // MARK: - Initialization
    
    @objc
    func initialize(_ apiKey: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async { [weak self] in
            guard let self = self else { return }
            
            // Set API key
            self.sdk.setApiKey(apiKey)
            
            // Subscribe to metrics updates
            self.sdk.$metricsBuffer
                .compactMap { $0 }
                .sink { [weak self] metrics in
                    self?.sendMetricsToJS(metrics)
                }
                .store(in: &self.cancellables)
            
            // Subscribe to edge metrics for face landmarks
            self.sdk.$edgeMetrics
                .compactMap { $0 }
                .sink { [weak self] edgeMetrics in
                    self?.sendEdgeMetricsToJS(edgeMetrics)
                }
                .store(in: &self.cancellables)
            
            // Subscribe to status updates
            self.vitalsProcessor.$statusHint
                .sink { [weak self] status in
                    self?.sendEvent(withName: "onStatusChange", body: ["status": status])
                }
                .store(in: &self.cancellables)
            
            self.isInitialized = true
            resolve(nil)
        }
    }
    
    // MARK: - Configuration
    
    @objc
    func setSmartSpectraMode(_ mode: String) {
        DispatchQueue.main.async { [weak self] in
            let spectraMode: SmartSpectraMode = mode.lowercased() == "spot" ? .spot : .continuous
            self?.sdk.setSmartSpectraMode(spectraMode)
        }
    }
    
    @objc
    func setCameraPosition(_ position: String) {
        DispatchQueue.main.async { [weak self] in
            let pos: AVCaptureDevice.Position = position.lowercased() == "back" ? .back : .front
            self?.sdk.setCameraPosition(pos)
        }
    }
    
    @objc
    func setMeasurementDuration(_ duration: Double) {
        DispatchQueue.main.async { [weak self] in
            self?.sdk.setMeasurementDuration(duration)
        }
    }
    
    @objc
    func setShowFps(_ show: Bool) {
        DispatchQueue.main.async { [weak self] in
            self?.sdk.setShowFps(show)
        }
    }
    
    @objc
    func setRecordingDelay(_ delay: Int) {
        DispatchQueue.main.async { [weak self] in
            self?.sdk.setRecordingDelay(delay)
        }
    }
    
    // MARK: - Measurement Control
    
    @objc
    func startMeasurement(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async { [weak self] in
            guard let self = self else { return }
            
            self.vitalsProcessor.startProcessing()
            self.vitalsProcessor.startRecording()
            
            self.sendEvent(withName: "onStatusChange", body: ["status": "measuring"])
            resolve(nil)
        }
    }
    
    @objc
    func stopMeasurement(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async { [weak self] in
            guard let self = self else { return }
            
            self.vitalsProcessor.stopRecording()
            self.vitalsProcessor.stopProcessing()
            
            self.sendEvent(withName: "onStatusChange", body: ["status": "ready"])
            resolve(nil)
        }
    }
    
    @objc
    func cleanup(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async { [weak self] in
            self?.cancellables.removeAll()
            self?.isInitialized = false
            resolve(nil)
        }
    }
    
    // MARK: - Data Conversion
    
    private func sendMetricsToJS(_ metrics: Presage_Physiology_MetricsBuffer) {
        var pulseRates: [[String: Any]] = []
        for rate in metrics.pulse.rate {
            pulseRates.append([
                "time": rate.time,
                "value": rate.value,
                "stable": rate.stable,
                "confidence": rate.confidence
            ])
        }
        
        var pulseTraces: [[String: Any]] = []
        for trace in metrics.pulse.trace {
            pulseTraces.append([
                "time": trace.time,
                "value": trace.value,
                "stable": trace.stable
            ])
        }
        
        var breathingRates: [[String: Any]] = []
        for rate in metrics.breathing.rate {
            breathingRates.append([
                "time": rate.time,
                "value": rate.value,
                "stable": rate.stable,
                "confidence": rate.confidence
            ])
        }
        
        var breathingUpperTraces: [[String: Any]] = []
        for trace in metrics.breathing.upperTrace {
            breathingUpperTraces.append([
                "time": trace.time,
                "value": trace.value,
                "stable": trace.stable
            ])
        }
        
        var blinkingStatus: [[String: Any]] = []
        for status in metrics.face.blinking {
            blinkingStatus.append([
                "time": status.time,
                "detected": status.detected,
                "stable": status.stable
            ])
        }
        
        var talkingStatus: [[String: Any]] = []
        for status in metrics.face.talking {
            talkingStatus.append([
                "time": status.time,
                "detected": status.detected,
                "stable": status.stable
            ])
        }
        
        var apneaStatus: [[String: Any]] = []
        for status in metrics.breathing.apnea {
            apneaStatus.append([
                "time": status.time,
                "detected": status.detected,
                "stable": status.stable
            ])
        }
        
        var inhaleExhaleRatios: [[String: Any]] = []
        for ratio in metrics.breathing.inhaleExhaleRatio {
            inhaleExhaleRatios.append([
                "time": ratio.time,
                "value": ratio.value,
                "stable": ratio.stable
            ])
        }
        
        let metricsData: [String: Any] = [
            "pulse": [
                "rate": pulseRates,
                "trace": pulseTraces
            ],
            "breathing": [
                "rate": breathingRates,
                "upperTrace": breathingUpperTraces,
                "lowerTrace": [],
                "amplitude": [],
                "apnea": apneaStatus,
                "respiratoryLineLength": [],
                "inhaleExhaleRatio": inhaleExhaleRatios
            ],
            "bloodPressure": [
                "phasic": []
            ],
            "face": [
                "blinking": blinkingStatus,
                "talking": talkingStatus
            ],
            "metadata": [
                "id": metrics.metadata.id,
                "uploadTimestamp": metrics.metadata.uploadTimestamp,
                "apiVersion": metrics.metadata.apiVersion
            ]
        ]
        
        sendEvent(withName: "onMetricsUpdate", body: metricsData)
    }
    
    private func sendEdgeMetricsToJS(_ edgeMetrics: Presage_Physiology_Metrics) {
        guard edgeMetrics.hasFace && !edgeMetrics.face.landmarks.isEmpty else { return }
        
        var landmarks: [[String: Any]] = []
        if let latestLandmarks = edgeMetrics.face.landmarks.last {
            for landmark in latestLandmarks.value {
                landmarks.append([
                    "x": landmark.x,
                    "y": landmark.y
                ])
            }
        }
        
        let edgeData: [String: Any] = [
            "face": [
                "landmarks": landmarks
            ]
        ]
        
        sendEvent(withName: "onEdgeMetricsUpdate", body: edgeData)
    }
}
*/

// Placeholder export for TypeScript
// The actual implementation requires native iOS setup
print("SmartSpectraModule.swift - Native iOS implementation for Presage SDK")
print("Follow the setup instructions in this file to integrate with your Expo bare workflow")
 

File: _legacy_archive/legacy_ref/atlas/native/android/SmartSpectraModule.kt
/**
 * Presage SmartSpectra SDK - Native Android Module
 * 
 * This Kotlin file needs to be added to your Android project when you eject from Expo.
 * It bridges the SmartSpectraSdk to React Native.
 * 
 * SETUP INSTRUCTIONS:
 * 1. Eject from Expo: npx expo prebuild
 * 2. Add to settings.gradle.kts:
 *    dependencyResolutionManagement {
 *        repositories {
 *            google()
 *            mavenCentral()
 *            maven { url = uri("https://jitpack.io") }
 *        }
 *    }
 * 3. Add to app/build.gradle.kts:
 *    dependencies {
 *        implementation("com.presagetech:smartspectra:1.0.26")
 *    }
 * 4. Update minSdk to 26 in build.gradle.kts
 * 5. Copy this file to android/app/src/main/java/com/aac/SmartSpectraModule.kt
 * 6. Register the module in MainApplication.kt
 * 7. Build and run
 */

/*
package com.aac

import android.util.Log
import androidx.camera.core.CameraSelector
import com.facebook.react.bridge.*
import com.facebook.react.modules.core.DeviceEventManagerModule
import com.presage.physiology.proto.MetricsProto.MetricsBuffer
import com.presage.physiology.proto.MetricsProto.Metrics
import com.presagetech.smartspectra.SmartSpectraMode
import com.presagetech.smartspectra.SmartSpectraSdk

class SmartSpectraModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    private val TAG = "SmartSpectraModule"
    private var sdk: SmartSpectraSdk = SmartSpectraSdk.getInstance()
    private var isInitialized = false

    override fun getName(): String = "SmartSpectraModule"

    private fun sendEvent(eventName: String, params: WritableMap?) {
        reactApplicationContext
            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
            .emit(eventName, params)
    }

    @ReactMethod
    fun initialize(apiKey: String, promise: Promise) {
        try {
            sdk.setApiKey(apiKey)
            
            // Set up metrics observer
            sdk.setMetricsBufferObserver { metricsBuffer ->
                sendMetricsToJS(metricsBuffer)
            }
            
            // Set up edge metrics observer for face landmarks
            sdk.setEdgeMetricsObserver { edgeMetrics ->
                sendEdgeMetricsToJS(edgeMetrics)
            }
            
            isInitialized = true
            promise.resolve(null)
        } catch (e: Exception) {
            Log.e(TAG, "Failed to initialize: ${e.message}")
            promise.reject("INIT_ERROR", e.message)
        }
    }

    @ReactMethod
    fun setSmartSpectraMode(mode: String) {
        val spectraMode = if (mode.uppercase() == "SPOT") {
            SmartSpectraMode.SPOT
        } else {
            SmartSpectraMode.CONTINUOUS
        }
        sdk.setSmartSpectraMode(spectraMode)
    }

    @ReactMethod
    fun setCameraPosition(position: String) {
        val pos = if (position.uppercase() == "BACK") {
            CameraSelector.LENS_FACING_BACK
        } else {
            CameraSelector.LENS_FACING_FRONT
        }
        sdk.setCameraPosition(pos)
    }

    @ReactMethod
    fun setMeasurementDuration(duration: Double) {
        sdk.setMeasurementDuration(duration)
    }

    @ReactMethod
    fun setShowFps(show: Boolean) {
        sdk.setShowFps(show)
    }

    @ReactMethod
    fun setRecordingDelay(delay: Int) {
        sdk.setRecordingDelay(delay)
    }

    @ReactMethod
    fun startMeasurement(promise: Promise) {
        try {
            // Note: In Android, measurement is typically controlled via SmartSpectraView
            // This is a simplified version - full implementation would use Activity callbacks
            val params = Arguments.createMap().apply {
                putString("status", "measuring")
            }
            sendEvent("onStatusChange", params)
            promise.resolve(null)
        } catch (e: Exception) {
            promise.reject("START_ERROR", e.message)
        }
    }

    @ReactMethod
    fun stopMeasurement(promise: Promise) {
        try {
            val params = Arguments.createMap().apply {
                putString("status", "ready")
            }
            sendEvent("onStatusChange", params)
            promise.resolve(null)
        } catch (e: Exception) {
            promise.reject("STOP_ERROR", e.message)
        }
    }

    @ReactMethod
    fun cleanup(promise: Promise) {
        try {
            isInitialized = false
            promise.resolve(null)
        } catch (e: Exception) {
            promise.reject("CLEANUP_ERROR", e.message)
        }
    }

    private fun sendMetricsToJS(metrics: MetricsBuffer) {
        val pulseRates = Arguments.createArray()
        metrics.pulse.rateList.forEach { rate ->
            val rateMap = Arguments.createMap().apply {
                putDouble("time", rate.time.toDouble())
                putDouble("value", rate.value.toDouble())
                putBoolean("stable", rate.stable)
                putDouble("confidence", rate.confidence.toDouble())
            }
            pulseRates.pushMap(rateMap)
        }

        val pulseTraces = Arguments.createArray()
        metrics.pulse.traceList.forEach { trace ->
            val traceMap = Arguments.createMap().apply {
                putDouble("time", trace.time.toDouble())
                putDouble("value", trace.value.toDouble())
                putBoolean("stable", trace.stable)
            }
            pulseTraces.pushMap(traceMap)
        }

        val breathingRates = Arguments.createArray()
        metrics.breathing.rateList.forEach { rate ->
            val rateMap = Arguments.createMap().apply {
                putDouble("time", rate.time.toDouble())
                putDouble("value", rate.value.toDouble())
                putBoolean("stable", rate.stable)
                putDouble("confidence", rate.confidence.toDouble())
            }
            breathingRates.pushMap(rateMap)
        }

        val breathingUpperTraces = Arguments.createArray()
        metrics.breathing.upperTraceList.forEach { trace ->
            val traceMap = Arguments.createMap().apply {
                putDouble("time", trace.time.toDouble())
                putDouble("value", trace.value.toDouble())
                putBoolean("stable", trace.stable)
            }
            breathingUpperTraces.pushMap(traceMap)
        }

        val blinkingStatus = Arguments.createArray()
        metrics.face.blinkingList.forEach { status ->
            val statusMap = Arguments.createMap().apply {
                putDouble("time", status.time.toDouble())
                putBoolean("detected", status.detected)
                putBoolean("stable", status.stable)
            }
            blinkingStatus.pushMap(statusMap)
        }

        val talkingStatus = Arguments.createArray()
        metrics.face.talkingList.forEach { status ->
            val statusMap = Arguments.createMap().apply {
                putDouble("time", status.time.toDouble())
                putBoolean("detected", status.detected)
                putBoolean("stable", status.stable)
            }
            talkingStatus.pushMap(statusMap)
        }

        val apneaStatus = Arguments.createArray()
        metrics.breathing.apneaList.forEach { status ->
            val statusMap = Arguments.createMap().apply {
                putDouble("time", status.time.toDouble())
                putBoolean("detected", status.detected)
                putBoolean("stable", status.stable)
            }
            apneaStatus.pushMap(statusMap)
        }

        val inhaleExhaleRatios = Arguments.createArray()
        metrics.breathing.inhaleExhaleRatioList.forEach { ratio ->
            val ratioMap = Arguments.createMap().apply {
                putDouble("time", ratio.time.toDouble())
                putDouble("value", ratio.value.toDouble())
                putBoolean("stable", ratio.stable)
            }
            inhaleExhaleRatios.pushMap(ratioMap)
        }

        val pulseMap = Arguments.createMap().apply {
            putArray("rate", pulseRates)
            putArray("trace", pulseTraces)
        }

        val breathingMap = Arguments.createMap().apply {
            putArray("rate", breathingRates)
            putArray("upperTrace", breathingUpperTraces)
            putArray("lowerTrace", Arguments.createArray())
            putArray("amplitude", Arguments.createArray())
            putArray("apnea", apneaStatus)
            putArray("respiratoryLineLength", Arguments.createArray())
            putArray("inhaleExhaleRatio", inhaleExhaleRatios)
        }

        val faceMap = Arguments.createMap().apply {
            putArray("blinking", blinkingStatus)
            putArray("talking", talkingStatus)
        }

        val metadataMap = Arguments.createMap().apply {
            putString("id", metrics.metadata.id)
            putString("uploadTimestamp", metrics.metadata.uploadTimestamp)
            putString("apiVersion", metrics.metadata.apiVersion)
        }

        val metricsData = Arguments.createMap().apply {
            putMap("pulse", pulseMap)
            putMap("breathing", breathingMap)
            putMap("bloodPressure", Arguments.createMap().apply {
                putArray("phasic", Arguments.createArray())
            })
            putMap("face", faceMap)
            putMap("metadata", metadataMap)
        }

        sendEvent("onMetricsUpdate", metricsData)
    }

    private fun sendEdgeMetricsToJS(edgeMetrics: Metrics) {
        if (!edgeMetrics.hasFace() || edgeMetrics.face.landmarksCount == 0) return

        val landmarks = Arguments.createArray()
        edgeMetrics.face.landmarksList.lastOrNull()?.let { latestLandmarks ->
            latestLandmarks.valueList.forEach { landmark ->
                val landmarkMap = Arguments.createMap().apply {
                    putDouble("x", landmark.x.toDouble())
                    putDouble("y", landmark.y.toDouble())
                }
                landmarks.pushMap(landmarkMap)
            }
        }

        val faceMap = Arguments.createMap().apply {
            putArray("landmarks", landmarks)
        }

        val edgeData = Arguments.createMap().apply {
            putMap("face", faceMap)
        }

        sendEvent("onEdgeMetricsUpdate", edgeData)
    }

    @ReactMethod
    fun addListener(eventName: String) {
        // Required for RN event emitter
    }

    @ReactMethod
    fun removeListeners(count: Int) {
        // Required for RN event emitter
    }
}
*/

// Placeholder - this file contains the Kotlin implementation as a comment
// The actual setup requires native Android project configuration
 
